federation/pkg/kubefed/init/init.go

```diff
@@ -470,6 +484,9 @@ func createService(cmdOut io.Writer, clientset client.Interface, namespace, svcN
 
 	var err error
 	svc, err = clientset.Core().Services(namespace).Create(svc)
+	if err != nil {
+		return svc, nil, nil, err
```

Comment:
perhaps return nil, nil, nil, err? I don't think you need to return the service if there was an error creating it.